### PR TITLE
Changing logging statments to Clayton's spec

### PR DESCRIPTION
**Pull request by jupierce**
Attempting to address concerns on #9247

fixes https://github.com/openshift/origin/issues/9524

### CODE REVIEW

**Comment by csrwng** [comment]
LGTM, but @bparees and/or @smarterclayton may also want to have a look

**Comment by smarterclayton** [comment]
Can you paste the output of a default build here so we can nit pick?

File: pkg/build/builder/sti.go [hunk 1]
```diff
@@ -227,7 +227,7 @@ func (s *S2IBuilder) Build() error {
 	config.PullAuthentication, _ = dockercfg.NewHelper().GetDockerAuth(config.BuilderImage, dockercfg.PullAuthType)
 	config.IncrementalAuthentication, _ = dockercfg.NewHelper().GetDockerAuth(pushTag, dockercfg.PushAuthType)
 
-	glog.V(0).Infof("Creating a new S2I builder with build config: %#v\n", describe.DescribeConfig(config))
+	glog.V(4).Infof("Creating a new S2I builder with build config: %#v\n", describe.DescribeConfig(config))
```
Comment (bparees) [inline pkg/build/builder/sti.go]: I understand the desire not to have this in the output, but this is the some of the most useful information we need when debugging user issues, so we're going to be asking everyone to recreate with a higher loglevel.

i suppose optimizing for failure isn't the right attitude, but i'm noting my objection anyway.

Comment (smarterclayton) [inline pkg/build/builder/sti.go]: Printing information on failures that lets you pin down the user problem is what we need to be doing.  Dumping the JSON doesn't help a user - printing real errors does.

Comment (mfojtik) [inline pkg/build/builder/sti.go]: agree with @smarterclayton we should write human-friendly describer

**Comment by bparees** [comment]
@smarterclayton @jupierce with these changes and the impending s2i bump, here's what the default build output looks like:

```
Downloading "https://github.com/openshift/ruby-hello-world.git" ...
---> Installing application source ...
---> Building your Ruby application from source ...
---> Running 'bundle install --deployment' ...
Fetching gem metadata from https://rubygems.org/..........
Installing rake 10.3.2
Installing i18n 0.6.11
Installing json 1.8.3
Installing minitest 5.4.2
Installing thread_safe 0.3.4
Installing tzinfo 1.2.2
Installing activesupport 4.1.7
Installing builder 3.2.2
Installing activemodel 4.1.7
Installing arel 5.0.1.20140414130214
Installing activerecord 4.1.7
Installing mysql2 0.3.16
Installing rack 1.5.2
Installing rack-protection 1.5.3
Installing tilt 1.4.1
Installing sinatra 1.4.5
Installing sinatra-activerecord 2.0.3
Using bundler 1.7.8
Your bundle is complete!
It was installed into ./bundle
---> Cleaning up unused ruby gems ...
Running post commit hook with image test/ruby-sample-build-1:b1250214 ...
/opt/rh/rh-ruby22/root/usr/bin/ruby -I"lib" -I"/opt/app-root/src/bundle/ruby/gems/rake-10.3.2/lib" "/opt/app-root/src/bundle/ruby/gems/rake-10.3.2/lib/rake/rake_test_loader.rb" "test/*_test.rb" 
Run options: --seed 65492

# Running:

.

Finished in 0.001518s, 658.5902 runs/s, 658.5902 assertions/s.

1 runs, 1 assertions, 0 failures, 0 errors, 0 skips
Pushing image 172.30.76.52:5000/test/origin-ruby-sample:latest ...
Pushed 0/10 layers, 9% complete
Pushed 1/10 layers, 17% complete
Pushed 2/10 layers, 25% complete
Pushed 3/10 layers, 33% complete
Pushed 4/10 layers, 43% complete
Pushed 5/10 layers, 51% complete
Pushed 6/10 layers, 62% complete
Pushed 7/10 layers, 72% complete
Pushed 8/10 layers, 86% complete
Pushed 9/10 layers, 100% complete
Pushed 10/10 layers, 100% complete
Push successful
```

**Comment by bparees** [comment]
do you want us to remove the initial Dowloading line too?

**Comment by smarterclayton** [comment]
I think we need downloading because it may take a while.

**Comment by smarterclayton** [comment]
Running post commit hook with image test/ruby-sample-build-1:b1250214 ...


Should just be "Running post-commit hook ..." (jessica and I talked today
about whether we want to come up with a "pretty printer" for the post
commit that handles very large hook values, and decided no for now).

**Comment by smarterclayton** [comment]
I *think* we may want a newline before the actual build log output is
printed and a line after.

**Comment by smarterclayton** [comment]
Line before "Pushing image"

**Comment by bparees** [comment]
at least some of that is more changes that need to go into s2i.  you're going to have to do a one-off if you want this in your screencap.

**Comment by bparees** [comment]
@jupierce can you update this w/ the parts of @smarterclayton's requests that apply to the origin logging so we can get it merged?

**Comment by jupierce** [comment]
Sure. Just pushed!

**Comment by bparees** [comment]
lgtm, thanks @jupierce 
@smarterclayton the rest of the changes you want will come from an s2i bump. 
[merge]

**Comment by bparees** [comment]
[testextended][extended:core(starting a build using CLI)]  - if this passes it will confirm that https://github.com/openshift/origin/issues/9524 is fixed by this PR.

**Comment by bparees** [comment]
woo (extended test success)

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/9557 
[test]

**Comment by bparees** [comment]
[test]

**Pull request merged**